pkgs/by-name/kr/krun/package.nix

```diff
@@ -0,0 +1,82 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, rustPlatform
+, cargo
+, libiconv
+, libkrun
+, makeWrapper
+, rustc
+, passt
+, sommelier
+, mesa
+, opengl-driver ? mesa.drivers
+, withSommelier ? false
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "krun";
+  version = "0.0.0-unstable-2024-06-18";
+
+  src = fetchFromGitHub {
+    owner = "slp";
+    repo = finalAttrs.pname;
+    rev = "912afa5c6525b7c8f83dffd65ec4b1425b3f7521";
+    hash = "sha256-rDuxv3UakAemDnj4Nsbpqsykts2IcseuQmDwO24L+u8=";
+  };
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/slp/krun/pull/39.diff";
+      hash = "sha256-CV69L+VDDLRcWgpgDCAYKLlTU9ytFcHhzNgOibWD8KY=";
+    })
+    (fetchpatch {
+      url = "https://github.com/slp/krun/pull/38.diff";
+      hash = "sha256-cK3iDhh+33H16V65lWUXahjmpSxI1HhiLUmkjfkRB7A=";
+    })
+  ];
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    hash = "sha256-efJvEpmSBTrShOc9fwmn2TqC1zYx5WO4LH/6JQtIErA=";
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
```

Comment:
So we normally just use `rustPlatform.buildRustPackage` (Is there anything wrong with that?)

Reply:
Wanted to use `finalAttrs` but it doesn't seem `buildRustPackage` supports that?

Reply:
Is `finalAttrs` really that important here? We don't use anything except `finalAttrs.src` which [does not work better than `rec`](https://github.com/NixOS/nixpkgs/issues/310373)

Reply:
Alright, I'll change it

Reply:
Migrated to using `buildRustPackage`